### PR TITLE
Add Scholarship and ScholarshipOportunity models

### PR DESCRIPTION
**Pull request by ReneSG**
What does this PR do?
- Deprecates Group and GroupUsers model in favor of Scholarship and ScholarshipOportunity
- Remove scholarship fields from user table to scholarship table.
- Updates all references to Group and GroupUses to user Scholarship and ScholarshipOportunity.

### CODE REVIEW

File: app/controllers/scholarship_oportunities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def show
+    @scholarship_oportunity = ScholarshipOportunity.find(params[:id])
+    @students = @scholarship_oportunity.students
+  end
+
+  def create
+    @scholarship_oportunity = ScholarshipOportunity.new(scholarship_oportunity_params)
+    @scholarship_oportunity.name = @scholarship_oportunity.name.parameterize(separator: '_')
+    if ScholarshipOportunity.exists?(name: @scholarship_oportunity.name)
```
Comment (eduardotru) [inline app/controllers/scholarship_oportunities_controller.rb:18]: Consider using `unless` if you're leaving the if branch empty

File: app/models/scholarship.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# Clase de  relacion m a n entre grupos y users
```
Comment (eduardotru) [inline app/models/scholarship.rb:3]: Consider updating the comments or removing them

File: app/models/scholarship_oportunity.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# Esta es la clase de convocatorias
```
Comment (eduardotru) [inline app/models/scholarship_oportunity.rb:3]: Ditto

File: app/models/scholarship_oportunity.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+class ScholarshipOportunity < ApplicationRecord
+  has_many :scholarship
```
Comment (eduardotru) [inline app/models/scholarship_oportunity.rb:5]: scholarship or scholarships?

Comment (ReneSG) [reply app/models/scholarship_oportunity.rb:5]: My bad.

File: app/models/scholarship.rb (hunk 2 above)
Comment (ReneSG) [reply app/models/scholarship.rb:3]: Done.

File: app/controllers/scholarship_oportunities_controller.rb (hunk 1 above)
Comment (ReneSG) [reply app/controllers/scholarship_oportunities_controller.rb:18]: Done

**Pull request merged**